nix/update.sh

```diff
@@ -51,3 +56,21 @@ update-source() {
 
 update-source caf "libvast/aux/caf"
 update-source fast_float "libvast/aux/fast_float"
+
+echo "Updating contrib/vast-plugins"
+vast_plugins_rev="$(get-submodule-rev "contrib/vast-plugins")"
+vast_plugins_json="$(jq --arg rev "${vast_plugins_rev#rev}" \
+  '."rev" = $rev' "${dir}/vast/plugins/source.json")"
+echo -E "${vast_plugins_json}" > "${dir}/vast/plugins/source.json"
```

Comment:
Ideally the workflow for updating this repo is just bumping the submodule. This looks like there are additional requirements. Can you document the workflow for updating the plugins?

Reply:
How about we fail the "VAST ... (Nix)" jobs with a nice message if it goes out of sync?

Reply:
Then the script should run without requiring Nix to be installed, the workflow should be documented, and the script should probably be in `/scripts` rather than `/nix`.

I don't understand why we would want to track versions for plugins in multiple places. This seems like an anti-pattern to me if we already have the submodule revision as a source of truth.

Reply:
As of this PR these numbers are meaningless. I only left them here to avoid touching the C++ code as well. As explained above it would be good to remove them in a separate PR.

Reply:
Okay if that is something you essentially want to do near-term, not so much if follow-up means medium- to long-term.

Reply:
I will do the follow up shortly after this one was merged. We need to come up with a solution for versioning 3rd party plugins though.